c/src/measureapi.cpp

```diff
@@ -37,8 +36,8 @@ struct tirexMeasureHandle_st final {
 		monitorthread.join();
 
 		// Stop measuring
-		for (auto& provider : providers | std::views::reverse)
-			provider->stop();
+		for (std::size_t i = providers.size() - 1; providers.size() > i; --i)
```

Comment:
I would personally prefer a [reverse iterator](https://en.cppreference.com/w/cpp/container/vector/rbegin) directly instead of relying on a random access container.